### ISSUE TITLE
collectFile() not resuming as expected

### ISSUE THREAD

**Issue by rulixxx** [nextflow-io/nextflow#3274]
## Bug report 

I'm building a list of vcfs into a file using collectFile() for input to a process. Sometimes the pipeline resumes correctly sometimes it doesn't. It appears to be related to the permanence of files in the /tmp directory. If the pipeline is resumed and a collect file is present in /tmp pipeline resumes correctly, otherwise pipeline will regenerate the collectFile and resume will fail.

I have tried the option tempDir in collectFile() to specify a permanent tmp dir location. However Nextflow removes this directory after completing. So a resume will again fail if /tmp is cleared of collect files.

### Expected behavior and actual behavior
Specified tempDir is not deleted and is used instead of /tmp so that resume works correctly.

### Steps to reproduce the problem

```
nextflow.enable.dsl=2

ch = Channel.of( "1","2","3" )

process CAT {
  input :
    path(file)

  script :
    """
    cat $file
    """
}
File tmp = new File("${workDir}/xxx")
tmp.mkdir()

workflow {
  file = ch.collectFile(name: 'numbers.txt',cache:'lenient', newLine: true, tempDir: "${workDir}/xxx")
  CAT(file)
}
```
-----
Remove collect files from /tmp (eg. /tmp/1a3a7f9956396439a70dbc9f4902bd9c.collect-file )
and attempt resume

### Program output 

Will run but will fail to resume

### Environment 

* Nextflow version: 22.04.5.5708
* Java version:  11.0.16
* Operating system: Linux
* Bash version: 4.4.20

### Additional context

(Add any other context about the problem here)

**Comment by pditommaso** [comment]
>  If the pipeline is resumed and a collect file is present in /tmp pipeline resumes correctly, otherwise pipeline will regenerate the collectFile and resume will fail.

this is the expected behavior

**Comment by rulixxx** [comment]
OK I understand...

However this causes problems when working on a cluster on a shared drive where Nextflow gets submitted itself as a job to the queue, as it will most likely get a new machine assigned. This will almost always doom a resume to fail.

I had to write my own collectFile processes to avoid the issue. not terrible, but it did take me several days to figure out the issue.


________________________________

**Comment by pditommaso** [comment]
Now I'm understanding it better. You are right it should not happen

**Comment by pditommaso** [comment]
Related to #3465

**Comment by bentsherman** [comment]
Resolved by #3466
